### PR TITLE
make public key field mandatory

### PR DESCRIPTION
**Pull request by xecdev**
This PR implements #89.

Test Plan:
- Install the plugin
- Check the 'Public Key' field in paywall settings

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

This PR makes the PayButton Public Key field mandatory in the Paywall Settings UI by adding the HTML5 `required` attribute and updating labels from "optional" to "required".

**Key Changes:**
- Added `required` attribute to public key input field
- Changed label text from "PayButton Public Key (optional)" to "PayButton Public Key (required)"
- Updated documentation from "strongly recommended" to "required"
- Added mention of "login transactions" alongside "paywall transactions"

**Critical Issue Found:**
- **Missing server-side validation**: The HTML5 `required` attribute can be easily bypassed (browser dev tools, disabled JS, direct API calls). The `save_settings()` method in `class-paybutton-admin.php` (line 295-298) saves the public key without checking if it's empty. This creates a security gap where users could save empty settings, causing payment verification to fail at runtime when `payment_trigger()` checks for the key (line 114-118 in `class-paybutton-ajax.php`).

<h3>Confidence Score: 2/5</h3>

- Not safe to merge - missing critical server-side validation for mandatory field
- Score reflects a critical security gap: the PR makes the field mandatory only at the UI level without server-side enforcement. Users can bypass HTML5 validation and save empty values, breaking payment verification at runtime. This defeats the purpose of making the field mandatory.
- templates/admin/paywall-settings.php needs server-side validation added to the save_settings() method in includes/class-paybutton-admin.php

<h3>Important Files Changed</h3>

File Analysis

| Filename | Score | Overview |
|----------|-------|----------|
| templates/admin/paywall-settings.php | 2/5 | Added HTML5 `required` attribute and updated labels, but missing critical server-side validation that could allow empty submissions |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Admin as WordPress Admin
    participant Form as Paywall Settings Form
    participant Server as save_settings()
    participant DB as WordPress Options
    participant Payment as payment_trigger()
    
    Note over Admin,Payment: Current Flow (Missing Validation)
    
    Admin->>Form: Save settings with empty public key
    Form->>Form: HTML5 validation (can be bypassed)
    Form->>Server: POST form data
    Server->>Server: sanitize_text_field()
    Note right of Server: ❌ No check if empty!
    Server->>DB: update_option('paybutton_public_key', '')
    DB-->>Server: Success
    Server-->>Admin: Settings saved
    
    Note over Admin,Payment: Later: Payment Attempt
    
    Payment->>Payment: Receive webhook from PayButton
    Payment->>DB: get_option('paybutton_public_key')
    DB-->>Payment: '' (empty string)
    Payment->>Payment: Check if empty
    Note right of Payment: ❌ Fails here at runtime!
    Payment-->>Payment: Return error
    
    Note over Admin,Payment: Should Add Server-Side Validation
    
    Admin->>Form: Save settings with empty public key
    Form->>Server: POST form data
    Server->>Server: Check if empty
    Note right of Server: ✓ Add validation here!
    Server-->>Admin: Error: Public key required
```

## Summary by CodeRabbit

* **New Features**
  * PayButton Public Key field is now required for configuration; applies to both paywall and login transactions.

* **Bug Fixes**
  * Enhanced validation to check for both wallet address and public key during setup.
  * Improved admin error notifications to clearly identify missing required settings.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by Klakurka** [comment]
@coderabbitai review

**Review by Klakurka — APPROVED** [review]
